receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/factory.go

```diff
@@ -54,6 +56,21 @@ func (f *Factory) CreateMetricsScraper(
 	config internal.Config,
 ) (scraperhelper.Scraper, error) {
 	cfg := config.(*Config)
+
+	if cfg.RootPath == "" {
+		for _, p := range []string{
+			"/.dockerenv",        // Mounted by dockerd when starting a container by default
+			"/run/.containerenv", // Mounted by podman as described here: https://github.com/containers/podman/blob/ecbb52cb478309cfd59cc061f082702b69f0f4b7/docs/source/markdown/podman-run.1.md.in#L31
+		} {
+			if _, err := os.Stat(p); !errors.Is(err, os.ErrNotExist) {
```

Comment:
Why do we want to show the warning on any error? I suppose there might be some permission errors that don't imply that those files exist. Should we check `err == nil` instead?

Reply:
You're right, sorry was trying to fly through all my open PRs.